ext/ldap/config.m4

```diff
@@ -79,7 +79,8 @@ if test "$PHP_LDAP" != "no"; then
     LDAP_PTHREAD=
   fi
 
-  MACHINE_INCLUDES=$($CC -dumpmachine)
+  dnl -pc removal is a hack for clang
+  MACHINE_INCLUDES=$($CC -dumpmachine | sed 's/-pc//')
```

Comment:
Seeing how most of the code below is 19 years old (https://github.com/php/php-src/blame/master/ext/ldap/config.m4#L85), I really wonder how much of it is really necessary. Searching for libssldap50 shows last mentions around ~2005...

Reply:
Regarding the ldap extension this is basically done to resolve a similar case as the one with gmp and there will be the same issue then as described in the comments at https://github.com/php/php-src/pull/4488 - there might be also `-unknown` used as a `$host_vendor`. So this `sed` solution above won't solve all possible cases.

Reply:
Right, it would be better to fix this properly as well. However, the library detection code here is very complex. I've submitted #4658 to clean out some parts.

Generally what is being done here is very suspicious, I don't think we do actual path-based checking for library files anywhere else. I'm thinking that this should be migrated to use something like PHP_CHECK_LIB.